File: 0560 - Subarray Sum Equals K/solution.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res, s, d = 0, 0, {0:1}
        for i in nums:
            s += i
            res += d.get(s-k, 0)
            d[s] = d.get(s, 0)+1
        return res

File: 0560 - Subarray Sum Equals K/solution.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int res = 0, s = 0;
        unordered_map<int, int> um{{0, 1}};
        for(auto i : nums){
            s += i;
            if(um.count(s-k)) res += um[s-k];
            um[s]++;
        }
        return res;
    }
};